src/bcf/report/report.rs

```diff
@@ -93,24 +93,22 @@ pub fn oncoprint(
                             continue;
                         }
 
-                        let impact = str::from_utf8(
-                            fields[*ann_indices.get(&String::from("IMPACT")).expect("No field named IMPACT found. Please only use VEP-annotated VCF-files.")],
-                        )?;
-                        let clin_sig = str::from_utf8(
-                            fields[*ann_indices.get(&String::from("CLIN_SIG")).expect("No field named CLIN_SIG found. Please only use VEP-annotated VCF-files.")],
-                        )?;
-                        let gene = str::from_utf8(
-                            fields[*ann_indices.get(&String::from("Gene")).expect("No field named Gene found. Please only use VEP-annotated VCF-files.")],
-                        )?;
-                        let dna_alteration = str::from_utf8(
-                            fields[*ann_indices.get(&String::from("HGVSc")).expect("No field named HGVSc found. Please only use VEP-annotated VCF-files.")],
-                        )?;
-                        let protein_alteration = str::from_utf8(
-                            fields[*ann_indices.get(&String::from("HGVSp")).expect("No field named HGVSp found. Please only use VEP-annotated VCF-files.")],
-                        )?;
-                        let consequence = str::from_utf8(
-                            fields[*ann_indices.get(&String::from("Consequence")).expect("No field named Consequence found. Please only use VEP-annotated VCF-files.")],
-                        )?;
+                        let get_field = |field: &str| {
+                            str::from_utf8(
+                                fields[*ann_indices.get(&field.to_owned()).expect(
+                                    &("No field named ".to_owned()
+                                        + field
+                                        + " found. Please only use VEP-annotated VCF-files."),
```

Comment:
Use format!(...) for better readibility.

Reply:
I was looking for something like that! Added with https://github.com/rust-bio/rust-bio-tools/pull/53/commits/ae2b9b34ad5a5121c9eeb2ac83cf45a59ff9d8dd.